### PR TITLE
Short form printing

### PR DESCRIPTION
**Pull request by mcabbott**
This adds a shorter printing for when a FillArray appears inside a tuple, or in an array.

Before:
```
julia> (1, Fill(ℯ,2), 3, Fill(π,4), 5)
(1, 2-element Fill{Irrational{:ℯ}}: entries equal to ℯ, 3, 4-element Fill{Irrational{:π}}: entries equal to π, 5)

julia> [Fill(i==j,2,2) for i in 1:3, j in 1:4]
3×4 Matrix{Fill{Bool, 2, Tuple{Base.OneTo{Int64}, Base.OneTo{Int64}}}}:
 2×2 Fill{Bool}: entries equal to true   2×2 Fill{Bool}: entries equal to false  2×2 Fill{Bool}: entries equal to false  2×2 Fill{Bool}: entries equal to false
 2×2 Fill{Bool}: entries equal to false  2×2 Fill{Bool}: entries equal to true   2×2 Fill{Bool}: entries equal to false  2×2 Fill{Bool}: entries equal to false
 2×2 Fill{Bool}: entries equal to false  2×2 Fill{Bool}: entries equal to false  2×2 Fill{Bool}: entries equal to true   2×2 Fill{Bool}: entries equal to false
 
julia> Fill(99,1)
1-element Fill{Int64}: entries equal to 99

julia> Fill(1.0, (3:4,))
2-element Fill{Float64, 1, Tuple{UnitRange{Int64}}} with indices 3:4: entries equal to 1.0
```
After:
```
julia> (1, Fill(ℯ,2), 3, Fill(π,4), 5)
(1, Fill(ℯ, 2), 3, Fill(π, 4), 5)

julia> [Fill(i==j,2,2) for i in 1:3, j in 1:4]
3×4 Matrix{Fill{Bool, 2, Tuple{Base.OneTo{Int64}, Base.OneTo{Int64}}}}:
 Fill(true, 2, 2)   Fill(false, 2, 2)  Fill(false, 2, 2)  Fill(false, 2, 2)
 Fill(false, 2, 2)  Fill(true, 2, 2)   Fill(false, 2, 2)  Fill(false, 2, 2)
 Fill(false, 2, 2)  Fill(false, 2, 2)  Fill(true, 2, 2)   Fill(false, 2, 2)

julia> Fill(99,1)
1-element Fill{Int64}, with entry equal to 99

julia> Fill(1.0, (3:4,))
2-element Fill{Float64, 1, Tuple{UnitRange{Int64}}} with indices 3:4, with entries equal to 1.0

julia> (Zeros{Int}(2), Ones{Bool}(3))  # should this print more?
(Zeros(2), Ones(3))

julia> Zygote.gradient((x,y) -> y * sum(x), ones(Float32, 3,3), pi)
(Fill(3.1415927f0, 3, 3), 9.0f0)
````

### CODE REVIEW

**Comment by dlfivefifty** [comment]
can you get the coverage up?

File: src/FillArrays.jl [hunk 1]
```diff
@@ -633,13 +655,12 @@ if VERSION ≥ v"1.5"
...
 end
 
-Base.show(io::IO, ::MIME"text/plain", x::Union{Eye,AbstractFill}) = show(io, x)
 
 ##
 # interface
 ##
 
-getindex_value(a::LinearAlgebra.AdjOrTrans) = getindex_value(parent(a))
+getindex_value(a::LinearAlgebra.AdjOrTrans) = getindex_value(parent(a)) # FillArrays.getindex_value(Adjoint(Fill(1+im,2,2)))
```
Comment (mcabbott) [inline src/FillArrays.jl:663]: I didn't mean to check this in, but think this is an unrelated bug -- adjoint is recursive.

Comment (dlfivefifty) [reply src/FillArrays.jl:663]: You are right. Can you fix this too and add a test?

Comment (mcabbott) [reply src/FillArrays.jl]: Sure, done!

**Comment by dlfivefifty** [comment]
Please bump the version too so I can tag...Do we consider this a breaking a change?

**Comment by mcabbott** [comment]
Your call, but I made it a patch for now. 

Some chance it breaks someone's doctests, I guess, but once they fix them they can bound the one version used for testing that. No functional change, and tagging a breaking change does impose some hassle on the other 90% of packages who don't have a doctest which prints this.

**Comment by dlfivefifty** [comment]
Great, if no one raises an objection I'll merge and tag tomorrow.

Thanks!

**Comment by dlfivefifty** [comment]
I think the overloaded ` map ` had a new return type? Maybe best to do v0.12

**Comment by mcabbott** [comment]
Sure, last commit is v0.12

**Comment by mcabbott** [comment]
Thanks! Maybe #147 can share the version bump too?
